satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -454,113 +454,151 @@ def is_high_resol(resolution):
-        nc.variables["channel_correlation_matrix_structured"].dims = ("channel_1", "channel_2")
-        self.nc = nc.drop_dims("channel")
+        for data_array in ds.data_vars.values():
+            data_array.attrs.pop("ancillary_variables", None)
 
-    def get_time(self):
-        """Get time coordinate.
 
-        Variable is sometimes named "time" and sometimes "time_ir_wv".
-        """
-        try:
-            return self["time_ir_wv"]
-        except KeyError:
-            return self["time"]
+class DatasetAccessor:
+    """Helper class for accessing the dataset."""
+
+    def __init__(self, ds):
+        """Wrap the given dataset."""
+        self.ds = ds
+
+    @property
+    def attrs(self):
+        """Exposes dataset attributes."""
+        return self.ds.attrs
+
+    def __getitem__(self, item):
+        """Get a variable from the dataset."""
+        data_array = self.ds[item]
+        if self._should_dims_be_renamed(data_array):
+            return self._rename_dims(data_array)
+        return data_array
+
+    def _should_dims_be_renamed(self, data_array):
+        """Determine whether dataset dimensions need to be renamed."""
+        return "y_ir_wv" in data_array.dims or "y_tie" in data_array.dims
+
+    def _rename_dims(self, data_array):
+        """Rename dataset dimensions to match satpy's expectations."""
+        new_names = {
+            "y_ir_wv": "y",
+            "x_ir_wv": "x",
+            "y_tie": "y",
+            "x_tie": "x"
+        }
+        new_names_avail = {
+            old: new
+            for old, new in new_names.items()
+            if old in data_array.dims
+        }
+        return data_array.rename(new_names_avail)
 
     def get_xy_coords(self, resolution):
         """Get x and y coordinates for the given resolution."""
         if is_high_resol(resolution):
-            return self.nc.coords["x"], self.nc.coords["y"]
-        return self.nc.coords["x_ir_wv"], self.nc.coords["x_ir_wv"]
+            return self.ds.coords["x"], self.ds.coords["y"]
+        return self.ds.coords["x_ir_wv"], self.ds.coords["x_ir_wv"]
 
     def get_image_size(self, resolution):
         """Get image size for the given resolution."""
         if is_high_resol(resolution):
-            return self.nc.coords["y"].size
-        return self.nc.coords["y_ir_wv"].size
+            return self.ds.coords["y"].size
+        return self.ds.coords["y_ir_wv"].size
+
+
+def open_dataset(filename):
+    """Load dataset from the given file."""
+    nc_raw = xr.open_dataset(
+        filename,
+        chunks={"x": CHUNK_SIZE,
+                "y": CHUNK_SIZE,
+                "x_ir_wv": CHUNK_SIZE,
+                "y_ir_wv": CHUNK_SIZE},
+        # see dataset preprocessor for why decoding is disabled
+        decode_cf=False,
+        decode_times=False,
+        mask_and_scale=False,
+    )
+    nc_preproc = DatasetPreprocessor().preprocess(nc_raw)
```

Comment:
How about having a function for that? I think it would look nicer with
```python
preprocessed_ds = preprocess(nc_raw)
```

Reply:
Good suggestion!